tests/test_ocr.py

```diff
@@ -1,105 +1,202 @@
     )
     request = OCRRequest(image=TEST_IMAGE_URL)
-    response = ocr_clarifai_client.request(request)
+    reference_response = OCRResponse(all_text=REFERENCE_OCR_TEXT, bounding_boxes=[])
+    response, metadata = ocr_clarifai_client.request(
+        request, reference=reference_response
+    )
 
     assert response is not None
     assert isinstance(response.all_text, str)
     assert len(response.all_text) > 0
     assert len(response.bounding_boxes) > 0
 
-    print_request_and_response(request, response, provider=Provider.CLARIFAI)
+    assert metadata is not None
+    assert isinstance(metadata, TaskMetadata)
+    assert metadata.task == Task.OCR
+    assert metadata.provider == Provider.CLARIFAI
+    assert metadata.latency > 0
+
+    print_request_response_and_metadata(request, response, metadata)
 
 
 def test_ocr_gcp():
-    ocr_gcp_client = supercontrast_client(task=Task.OCR, providers=[Provider.GCP])
+    ocr_gcp_client = SuperContrastClient(task=Task.OCR, providers=[Provider.GCP])
     request = OCRRequest(image=TEST_IMAGE_URL)
-    response = ocr_gcp_client.request(request)
+    reference_response = OCRResponse(all_text=REFERENCE_OCR_TEXT, bounding_boxes=[])
+    response, metadata = ocr_gcp_client.request(request, reference=reference_response)
 
     assert response is not None
     assert isinstance(response.all_text, str)
     assert len(response.all_text) > 0
     assert len(response.bounding_boxes) > 0
 
-    print_request_and_response(request, response, provider=Provider.GCP)
+    assert metadata is not None
+    assert isinstance(metadata, TaskMetadata)
+    assert metadata.task == Task.OCR
+    assert metadata.provider == Provider.GCP
+    assert metadata.latency > 0
+
+    print_request_response_and_metadata(request, response, metadata)
 
 
 def test_ocr_sentisight():
-    ocr_sentisight_client = supercontrast_client(
+    ocr_sentisight_client = SuperContrastClient(
         task=Task.OCR, providers=[Provider.SENTISIGHT]
     )
     request = OCRRequest(image=TEST_IMAGE_URL)
-    response = ocr_sentisight_client.request(request)
+    reference_response = OCRResponse(all_text=REFERENCE_OCR_TEXT, bounding_boxes=[])
+    response, metadata = ocr_sentisight_client.request(
+        request, reference=reference_response
+    )
 
     assert response is not None
     assert isinstance(response.all_text, str)
     assert len(response.all_text) > 0
     assert len(response.bounding_boxes) > 0
 
-    print_request_and_response(request, response, provider=Provider.SENTISIGHT)
+    assert metadata is not None
+    assert isinstance(metadata, TaskMetadata)
+    assert metadata.task == Task.OCR
+    assert metadata.provider == Provider.SENTISIGHT
+    assert metadata.latency > 0
+
+    print_request_response_and_metadata(request, response, metadata)
+
+
+# evaluate
+
+
+def test_ocr_evaluate():
+    ocr_client = SuperContrastClient(
+        task=Task.OCR,
+        providers=[
```

Comment:
it would be cool if we had a function which lists all the providers for a task

Reply:
I added functions for checking both directions and a test to make sure it works.